test/test_autotag.py

```diff
@@ -560,8 +560,8 @@ def test_order_works_with_missing_tracks(self):
             items, trackinfo
         )
         assert extra_items == []
-        assert extra_tracks == [trackinfo[1]]
-        assert mapping == {items[0]: trackinfo[0], items[1]: trackinfo[2]}
+        assert extra_tracks == [trackinfo[2]]
+        assert mapping == {items[0]: trackinfo[0], items[1]: trackinfo[1]}
```

Comment:
I think I understand why this change was made (the original code didn't fail the test due to indexing the arrays from the back with the spurious -1 index), and this made the test fail.

However, the original test case seems to cover something slightly different (track assignment with missing tracks with gaps, i.e. just `zip(track_infos, items)` (stopping when the shortest argument ends) wouldn't give the correct assignment.

Maybe, it would make sense to enhance the test case such that it tests missing tracks at various positions:

```py
    def test_order_works_with_missing_tracks(self):
    	all_items = [
            self.item("one", 1),
            self.item("two", 2),
            self.item("three", 3),
        ]
        all_trackinfos = [
            TrackInfo(title="one"),
            TrackInfo(title="two"),
            TrackInfo(title="three"),
        ]
    	for missing_idx in range(3):
            items = all_items[:]
            items.pop(missing_idx)
            trackinfos = all_trackinfos[:]
            trackinfos.pop(missing_idx)
            mapping, extra_items, extra_tracks = match.assign_items(
                items, all_trackinfos
            )
            assert extra_items == []
            assert extra_tracks == [trackinfo[missing_idx]]
            assert mapping == dict(zip(items, trackinfos))
```

I didn't test this, but this should contain both the original and your modified test case.

Reply:
I refactored the tests and added the test cases that you mentioned above